### PR TITLE
landing worker: fix failed paths parsing (bug 1696396)

### PR DESCRIPTION
**Pull request by zzzeid**
- do not fetch failed paths form raw output
    - look at .rej paths instead
- trim last 4 characters instead of using .rstrip to get file path

### CODE REVIEW

File: landoapi/landing_worker.py [hunk 1]
```diff
@@ -345,7 +340,7 @@ def run_job(self, job, repo, hgrepo, treestatus, patch_bucket):
                                     reject["content"] = f.read()
                             except Exception as e:
                                 logger.exception(e)
-                            breakdown["reject_paths"][r.rstrip(".rej")] = reject
+                            breakdown["reject_paths"][r[:-4]] = reject
```
Comment (cgsheeh) [inline landoapi/landing_worker.py:343]: Let's add a comment explaining the significance of `:-4` here.

File: landoapi/landing_worker.py [hunk 2]
```diff
@@ -227,18 +227,12 @@ def extract_error_data(exception: str) -> tuple[list[str], list[str]]:
...
-        # RE to capture files that could not be merged for one reason or another.
-        patterns = (
-            "(?:patching file (.*))",
-            "(?:file (.*) already exists)",
-            "(?:unable to find '(.*)' for patching)",
-        )
+        # TODO: capture reason for patch failure, e.g. deleting non-existing file, or
+        # adding a pre-existing file, etc...
+        reject_paths = rejs_re.findall(exception)
+        failed_paths = [path[:-4] for path in reject_paths]
```
Comment (cgsheeh) [inline landoapi/landing_worker.py:233]: Same here.

File: landoapi/landing_worker.py [hunk 3]
```diff
@@ -227,18 +227,12 @@ def extract_error_data(exception: str) -> tuple[list[str], list[str]]:
             re.MULTILINE,
         )
 
-        # RE to capture files that could not be merged for one reason or another.
```
Comment (cgsheeh) [inline landoapi/landing_worker.py:230]: If I understand correctly, we have some error messages interspersed with some confirmation/successful messages, and our workaround to fix the problem is to stop parsing for both. Could we (or do we plan to) re-enable this parsing and account for the potential "successfully applied hunk" error message?

**Review by cgsheeh — CHANGES_REQUESTED** [review]

File: landoapi/landing_worker.py (hunk 3 above)
Comment (zzzeid) [reply landoapi/landing_worker.py:230]: Yes - I'll file a bug for this. That particular functionality would be needed in order to determine that cause of the conflict, but not specifically to find the failed paths (which can be done just by looking at the .rej paths).

**Review by cgsheeh — APPROVED** [review]

**Pull request merged**